File: clients/src/main/java/org/yigitcanyontem/clients/users/profile/UsersProfileDto.java
package org.yigitcanyontem.clients.users.profile;

import lombok.*;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UsersProfileDto {
    private Integer id;
    private Integer usersId;
    private String firstName;
    private String lastName;
    private String profilePictureUrl;
    private String bio;
    private String city;
    private String country;
    private String website;
    private String jobTitle;
    private Date birthDate;
    private Date createdAt;
    private String linkedinProfile;
    private String githubProfile;
    private String mediumProfile;
}

File: frontend/codeConnect/src/models/user/UsersProfileDto.ts
export interface UsersProfileDto {
    id: number;
    usersId: number;
    firstName: string;
    lastName: string;
    profilePictureUrl: string;
    bannerPictureUrl: string;
    bio: string;
    city: string;
    country: string;
    website: string;
    jobTitle: string;
    birthDate: Date;
    createdAt: Date;
    linkedinProfile: string;
    githubProfile: string;
    mediumProfile: string;
    followingCount: number;
    followersCount: number;
}
